vignettes/irecist.Rmd

```diff
@@ -0,0 +1,858 @@
+
+In this case we use the overall response records from `RS` from the
+investigator as our starting point. The parameter details such as
+`PARAMCD`, `PARAM` etc will always be company-specific, but an example
+is shown below so that you can trace through how these records feed into
+the other parameter derivations.
+
+```{r}
+adrs <- adrs %>%
+  filter(RSEVAL == "INVESTIGATOR" & RSTESTCD == "OVRLRESP") %>%
+  mutate(
+    PARAMCD = "OVR",
+    PARAM = "Overall Response by Investigator",
+    PARCAT1 = "Tumor Response",
+    PARCAT2 = "Investigator",
+    PARCAT3 = "iRecist"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, VISIT, RSTESTCD, RSEVAL, PARAMCD, PARAM, PARCAT1, PARCAT2, PARCAT3)
+)
+```
+
+### Partial Date Imputation and Deriving `ADT`, `ADTF`, `AVISIT` etc
+
+If your data collection allows for partial dates, you could apply a
+company-specific imputation rule at this stage when deriving `ADT`. For
+this example, here we impute missing day to last possible date.
+
+```{r}
+adrs <- adrs %>%
+  derive_vars_dt(
+    dtc = RSDTC,
+    new_vars_prefix = "A",
+    highest_imputation = "D",
+    date_imputation = "last"
+  ) %>%
+  mutate(AVISIT = VISIT)
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, RSDTC, ADT, ADTF)
+)
+```
+
+### Derive `AVALC` and `AVAL`
+
+Here we populate `AVALC` and create the numeric version as `AVAL`
+(ordered from best to worst response). The `AVAL` values are not considered in
+the parameter derivations below, and so changing `AVAL` here would not change
+the result of those derivations.
+
+iRECIST ordering will be used or if you'd like to provide your own company-specific ordering here you
+could do this as follows:
+
+```{r}
+aval_resp_new <- function(arg) {
+  case_when(
+    arg == "iCR" ~ 8,
+    arg == "iPR" ~ 7,
+    arg == "iSD" ~ 6,
+    arg == "NON-iCR/NON-iUPD" ~ 5,
+    arg == "iCPD" ~ 4,
+    arg == "iUPD" ~ 3,
+    arg == "NE" ~ 2,
+    arg == "MISSING" ~ 1,
+    TRUE ~ NA_real_
+  )
+}
+
+adrs <- adrs %>%
+  mutate(
+    AVALC = RSSTRESC,
+    AVAL = aval_resp_new(AVALC)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, RSSTRESC, AVALC, AVAL)
+)
+```
+
+### Flag Worst Assessment at Each Date (`ANL01FL`) {#anl01fl}
+
+When deriving `ANL01FL` this is an opportunity to exclude any records
+that should not contribute to any downstream parameter derivations. In
+the below example this includes only selecting valid assessments and
+those occurring on or after randomization date. If there is more than
+one assessment at a date, the worst one is flagged.
+
+```{r}
+adrs <- adrs %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = exprs(STUDYID, USUBJID, ADT),
+      order = exprs(AVAL, RSSEQ),
+      new_var = ANL01FL,
+      mode = "last"
```

Comment:
This is flagging the best instead of the worst result. I would also ignore results with `AVALC == "MISSING"`.

Reply:
will update

Reply:
in our case mode should equal to "first", what about NE and and iCPD responses on the same day?